src/xkbcomp/keymap-dump.c

```diff
@@ -88,7 +88,8 @@ check_write_buf(struct buf *buf, const char *fmt, ...)
 
     available = buf->alloc - buf->size;
     va_start(args, fmt);
-    printed = vsnprintf(buf->buf + buf->size, available, fmt, args);
+    printed = vsnprintf(buf->buf ? buf->buf + buf->size : buf->buf, available,
```

Comment:
Is `buf->buf` really meant to be NULL here?

Reply:
It can if `available` is 0. We try to write or fail with the required size used later to reallocate. I added a comment.